### ISSUE TITLE
[BUG] `CIPHER_LIST` on Windows is wrong since in v0.6.x

### ISSUE THREAD

**Issue by ed0nkey** [yifeikong/curl-impersonate#54]
While using `impersonate="safari17_2_ios"`, I get different JA3 digest results between my Windows and macOS machines (`8be0b641abb257fae7b13bcfd2657032` on Mac and `a76d766e1e01aa4cfaee1331b1bada3b` on Windows).
Unfortunately this issue is triggering cloudflare on the Windows machine while on Mac it works just fine every time. 
Just for testing I've tried using different python versions with an older openssl version but that is not the problem as the digest doesn't change.

**Comment by yifeikong** [comment]
I may have found the root cause, it should be a side effect of #12, when we enabled `OPENSSL_NO_ASM` on Windows

If `OPENSSL_NO_ASM` is defined, then `int hwaes_capable()` will return `0`, which means `SSL_CTX_set_cipher_list` will put `chachaciphers`(4867) in the first place, hence the wrong order.

```cpp
size_t num = 0;                                         
if (has_aes_hw) {                                       
  for (uint16_t id : kAESCiphers) {                     
    co_list[num++].cipher = SSL_get_cipher_by_value(id);
    assert(co_list[num - 1].cipher != nullptr);         
  }                                                     
}                                                       
for (uint16_t id : kChaChaCiphers) {                    
  co_list[num++].cipher = SSL_get_cipher_by_value(id);  
  assert(co_list[num - 1].cipher != nullptr);           
}                                                       
if (!has_aes_hw) {                                      
  for (uint16_t id : kAESCiphers) {                     
    co_list[num++].cipher = SSL_get_cipher_by_value(id);
    assert(co_list[num - 1].cipher != nullptr);         
  }                                                     
}                                                                                                     
```

**Comment by gamer191** [comment]
Based on the issue title, does 0.5.10 still have correct fingerprints?

**Comment by yifeikong** [comment]
> Based on the issue title, does 0.5.10 still have correct fingerprints?

Yes.

**Comment by yifeikong** [comment]
@T-256 There is no build yet.

**Comment by yifeikong** [comment]
Please try the new version 0.7.0b1 [here](https://github.com/yifeikong/curl-impersonate/releases)

**Comment by yifeikong** [comment]
@gamer191 The [CVE-2023-38545](https://github.com/advisories/GHSA-7xw9-w465-6x42) was also fixed by upgrading to 8.5.0 of curl.

**Comment by ed0nkey** [comment]
Can confirm it works for me, thank you 👍
